Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,160 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int FindDigit(string equation)
+        {
+            /*
+            I have solved the problem in the following manner :
+            1) extract the operands
+            2) find which one has the missing digit
+            3) find the actual number that would make the equation work as expected
+            4) replace the missing digit by comparing the string to actual result
+            5) return the missing digit if the result matches the operand after replacing the missing digit
+             */
+
+
+            // extracting the numbers out of the equation.
+            string[] num = new String[3];
+            int[] number = new int[3];
+            int questionMarkIndex= -1; // to store which operand has the missing digit.
+            int index=0;
+            int result=-1; // to store the character that is to be replaced by ?
+            string temp; // to store the actual retrieved number that has the 'question mark'
+
+            for(int i=0; i<equation.Length; i++){
+                if(equation[i]=='*'||equation[i]=='='){
+                    index++;
+                }else{
+                    num[index] += equation[i];
+                }
+            }
+            
+            // now let's find question mark index, to check which number has tha missing digit!
+            // and store the information in the variable questionMarkIndex
+            if(num[0].Contains('?')){
+                questionMarkIndex = 0;
+            }else if(num[1].Contains('?')){
+                questionMarkIndex = 1;
+            }else if(num[2].Contains('?')){
+                questionMarkIndex = 2;
+            }
+
+            if(questionMarkIndex==0){
+                //Console.WriteLine("missing number is First");
+                // Now we retieve the number missing!
+                number[1]= Int32.Parse(num[1]);

```

Comment:
`Int32.TryParse` is better option than `Int32.Parse`

Reply:
I read the difference between the two and made the changes to Int32.TryParse